src/main/java/github/scarsz/discordsrv/api/SlashCommandManager.java

```diff
@@ -0,0 +1,142 @@
+        }
+        printSlashRegistrationError(errors, commandsMap.keySet());
+    }
+
+    /**
+     * Sets the slash commands for a specific plugin, overrides old ones. this method does not register them, see {@link SlashCommandManager#reloadSlashCommands()}
+     * @param plugin Plugin that is registering the commands, whenever discordsrv is registering the commands it will not register the commands for that plugin if it is disabled
+     * @param commandData Commands to add
+     */
+    public void setCommands(@NotNull Plugin plugin, @NotNull CommandData... commandData) {
+        PluginCommands pluginCommands = commandsMap.get(plugin);
+        if (pluginCommands == null) {
+            pluginCommands = new PluginCommands(plugin, new HashSet<>(Arrays.asList(commandData)));
+            commandsMap.put(plugin, pluginCommands);
+        } else pluginCommands.data = new HashSet<>(Arrays.asList(commandData));
+    }
+
+    private void printSlashRegistrationError(Map<Guild, RegistrationResult> errors, Set<Plugin> plugins) {
+        if (errors.size() == (int) errors.values().stream().filter(r -> r == RegistrationResult.RATE_LIMIT).count()) {
+            DiscordSRV.getPlugin().getLogger().warning("Rate limited while registering in the following guilds: " + errors.keySet().stream().map(Guild::getName).collect(Collectors.joining(", ")));
+            return;
+        }
+        DiscordSRV.getPlugin().getJda().setRequiredScopes("applications.commands");
+        String invite = DiscordSRV.getPlugin().getJda().getInviteUrl();
+        DiscordSRV.getPlugin().getLogger().warning("==============================================================");
+        DiscordSRV.getPlugin().getLogger().warning("DiscordSRV could not register slash commands to some discord servers!");
+        for (Guild guild : errors.keySet()) {
+            RegistrationResult result = errors.get(guild);
+            switch (result) {
+                case MISSING_SCOPE:
+                    DiscordSRV.getPlugin().getLogger().warning("Missing scopes in " + guild.getName() + " (" + guild.getId() + ")");
+                case UNKNOWN_ERROR:
+                    DiscordSRV.getPlugin().getLogger().warning("Unknown error in " + guild.getName() + " (" + guild.getId() + ")");
+                case RATE_LIMIT:
+                    DiscordSRV.getPlugin().getLogger().warning("Rate limited in " + guild.getName() + " (" + guild.getId() + ")");
+            }
+        }
+        if (errors.values().stream().anyMatch(r -> r == RegistrationResult.MISSING_SCOPE)) DiscordSRV.getPlugin().getLogger().warning("Use " + invite + " to re-invite the bot to guilds with missing scope!");
+        DiscordSRV.getPlugin().getLogger().warning(" ");
+        DiscordSRV.getPlugin().getLogger().warning(plugins.size() == 1 && plugins.toArray(new Plugin[0])[0].equals(DiscordSRV.getPlugin()) ?
```

Comment:
```suggestion
        DiscordSRV.getPlugin().getLogger().warning(plugins.size() == 1 && plugins.get(0).equals(DiscordSRV.getPlugin()) ?
```

And stop spamming `DiscordSRV.getPlugin().getLogger()`